File: Arrays/Longest_Common_Prefix/vertical_scan.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        s = ""
        min_length = min(len(word) for word in strs)
        for p in range(min_length):
            q = strs[0][p]
            mark = True
    
            for y in range(1,len(strs)):
                if strs[y][p] == q:
                    continue
                else:
                    mark = False
                    break
    
            if mark:
                s += q
            else:
                break
        return s

File: Arrays/Longest_Common_Prefix/vertical_scan.cpp
#include <vector>
#include <string>
#include <algorithm>
using namespace std;

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) return "";
        
        size_t minLen = strs[0].size();
        for (const string& str : strs) {
            minLen = min(minLen, str.size());
        }
        
        for (size_t i = 0; i < minLen; i++) {
            char ch = strs[0][i];
            for (size_t j = 1; j < strs.size(); j++) {
                if (strs[j][i] != ch) {
                    return strs[0].substr(0, i);
                }
            }
        }
        return strs[0].substr(0, minLen);
    }
};
